eclair-core/src/test/scala/fr/acinq/eclair/blockchain/watchdogs/BlockchainWatchdogSpec.scala

```diff
@@ -86,4 +93,34 @@ class BlockchainWatchdogSpec extends ScalaTestWithActorTestKit(ConfigFactory.loa
     eventListener.expectNoMessage(100 millis)
   }
 
+  test("fetch block headers on mainnet over Tor", TestTags.ExternalApi) {
+    val proxyParams = Socks5ProxyParams(new InetSocketAddress("127.0.0.1", 9050),
+      credentials_opt = None,
+      randomizeCredentials = true,
+      useForIPv4 = true,
+      useForIPv6 = true,
+      useForTor = true)
+
+    if (Socks5ProxyParams.proxyAcceptsConnections(proxyParams)) {
+      val eventListener = TestProbe[DangerousBlocksSkew]()
+      system.eventStream ! EventStream.Subscribe(eventListener.ref)
+
+      val nodeParams = TestConstants.Alice.nodeParams.copy(
+        chainHash = Block.LivenetGenesisBlock.hash,
+        socksProxy_opt = Some(proxyParams))
+      val watchdog = testKit.spawn(BlockchainWatchdog(nodeParams, 1 second))
+      watchdog ! WrappedCurrentBlockCount(630561)
+
+      val events = Seq(
+        eventListener.expectMessageType[DangerousBlocksSkew],
+        eventListener.expectMessageType[DangerousBlocksSkew],
+        eventListener.expectMessageType[DangerousBlocksSkew]
+      )
+      eventListener.expectNoMessage(100 millis)
+      assert(events.map(_.recentHeaders.source).toSet === Set("blockstream.info", "mempool.space"))
```

Comment:
That doesn't look right, have you run this test (disclaimer: I haven't yet)?
How can there be 3 events with only 2 watchdogs?

Reply:
Got it, disabling `blockcypher` for this test is OK.
In that case you only have two watchdogs, so you should only expect two `DangerousBlocksSkew` events, not four (when defining `events` above).
Expecting four works as well because after the `blockTimeout`, we will call again the two watchdogs, but it makes the test wait for no reason.